### PR TITLE
UCT/BUILD: fixed config.h include

### PR DESCRIPTION
**Pull request by hoopoepg**
- fixed missing and incorrect include config.h

### CODE REVIEW

File: src/uct/base/uct_iface.h [hunk 1]
```diff
@@ -7,6 +7,10 @@
 #ifndef UCT_IFACE_H_
 #define UCT_IFACE_H_
 
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
```
Comment (yosefe) [inline src/uct/base/uct_iface.h:12]: why needed in this H file?
is it to not have to include it from all transports?
seems like a "band-aid" ...

File: src/uct/tcp/tcp_net.c [hunk 2]
```diff
@@ -5,12 +5,12 @@
  * See file LICENSE for terms.
  */
 
-#include "tcp.h"
-
 #ifdef HAVE_CONFIG_H
 #  include "config.h"
 #endif
 
+#include "tcp.h"
```
Comment (yosefe) [inline src/uct/tcp/tcp_net.c:12]: 👍

File: src/uct/base/uct_iface.h (hunk 1 above)
Comment (hoopoepg) [reply src/uct/base/uct_iface.h:12]: it needed in all files where macro like ```IBV_HW_TM``` is used (on line 44 below)

**Comment by hoopoepg** [comment]
@yosefe ok to merge?

**Review by yosefe — APPROVED** [review]

**Pull request merged**